pkg/api/api.go

```diff
@@ -169,3 +173,36 @@ func (a API) ReceiveEvent(w http.ResponseWriter, r *http.Request) {
 		Status: 200,
 	})
 }
+
+// Invoke creates an event to invoke a specific function.
+func (a API) Invoke(w http.ResponseWriter, r *http.Request) {
+	// XXX: In OSS self hosting, check signing keys here.
```

Comment:
for future discussions:
this should really be API keys at this point. using signing keys for authn / authz is expanding the use case too much.

Reply:
I'm fairly certain that signing keys will become environment-scoped API keys, but of course open to feedback and ideas.  Having three keys seems excessive, plus APIs like this are always scoped by environment to "invoke" a function within your env.